cmd/warpforge/main_test.go

```diff
@@ -82,12 +94,29 @@ func cleanRunRecord(str string) string {
 	return strings.TrimSpace(str)
 }
 
-func execFn(args []string, stdin io.Reader, stdout, stderr io.Writer) (int, error) {
-	err := makeApp(stdin, stdout, stderr).Run(args)
-	if err != nil {
-		return 1, err
+func buildExecFn(projPath string) func(args []string, stdin io.Reader, stdout io.Writer, stderr io.Writer) (int, error) {
+	return func(args []string, stdin io.Reader, stdout, stderr io.Writer) (int, error) {
+		// override the path to required binaries
+		err := os.Setenv("WARPFORGE_PATH", filepath.Join(projPath, "plugins"))
+
+		// set up a root workspace in the testmark run directory
+		testmarkWd, err := os.Getwd()
+		if err != nil {
+			return 1, fmt.Errorf("failed to get testmark pwd: %s", err)
+		}
+		os.MkdirAll(filepath.Join(testmarkWd, ".warpforge/catalogs/default"), 0755)
+		copyCmd := exec.Command("cp", "--recursive", filepath.Join(projPath, ".warpforge", "catalog"), filepath.Join(testmarkWd, ".warpforge", "catalogs"))
+		copyCmd.Run()
+		copyCmd = exec.Command("cp", "--recursive", filepath.Join(projPath, ".warpforge", "warehouse"), filepath.Join(testmarkWd, ".warpforge"))
+		copyCmd.Run()
```

Comment:
Can we get rid of this copy operation entirely by adding a feature where an env var like `WARPFORGE_EXTRA_WAREHOUSEADDR=file+ca:///over/here` will just let us get this from where it already resides in the filesystem?